### PR TITLE
Carrierwave CVE Fix

### PR DESCRIPTION
**Pull request by eanders**
[//]: # 'remove this if PR is for a release-* branch'
# _Please squash merge this PR_

## Description

This implements the suggested work-around for Carrier wave for CVE-2023-49090 since no 1.x version is provided.

## Type of change
[//]: # 'remove options that are not relevant'
- [x] Dependency update

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (rubocop)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: config/initializers/carrierwave.rb [hunk 1]
```diff
@@ -17,3 +17,16 @@
...
 end
+
+# FIX for CVE-2023-49090
+module CarrierWave
+  module Uploader
+    module ContentTypeWhitelist
+      private
+
+      def whitelisted_content_type?(content_type)
+        Array(content_type_whitelist).any? { |item| content_type =~ /\A#{item}/ }
```
Comment (yakloinsteak) [inline config/initializers/carrierwave.rb:29]: Is there any possibility of content_type being nil? Maybe `to_s` if we're unsure?

When I monkey-patch, I sometimes raise an exception if the gem version isn't exactly the version that we're monkey-patching.

**Review by yakloinsteak — APPROVED** [review]

**Review by awisema — APPROVED** [review]

File: config/initializers/carrierwave.rb (hunk 1 above)
Comment (eanders) [reply config/initializers/carrierwave.rb:29]: @yakloinsteak does line 22 do what you were thinking?   It appears to error if the version doesn't match.

Comment (yakloinsteak) [reply config/initializers/carrierwave.rb:29]: Yeah, that looks good.

Comment (eanders) [reply config/initializers/carrierwave.rb:29]: Thanks!

**Pull request merged**